designer/client/src/components/modals/DeployWithParametersDialog.tsx

```diff
@@ -132,7 +130,17 @@ export function DeployWithParametersDialog(props: WindowContentProps<WindowKind,
                             {t("dialog.advancedParameters.title", "Advanced parameters")}
                         </Typography>
                         {Object.entries(groupBy(parametersDefinition, (def) => def.componentId)).map(([componentId, nodeParameters]) => (
-                            <AdvancedParametersSection key={componentId} componentId={componentId}>
+                            <AdvancedParametersSection
+                                key={componentId}
+                                componentId={componentId}
+                                expanded={expandedState[componentId]}
+                                onChange={(isExpanded) =>
+                                    setExpandedState({
```

Comment:
It can be problematic in a react because in some situations it can cause stale state issues, like a lot of component updates or delays. To be sure, that you always use the up-to-date state, use previousState

```
setExpandedState(prevState => ({
    ...prevState,
    [componentId]: isExpanded,
}));
```

in this case, you are sure, that the up to date state is use

Reply:
Done